.github/workflows/build-with-bal-test-native.yml

```
@@ -17,6 +17,14 @@ on:
                 default: ''
     schedule:
         -   cron: '30 18 * * *'
+    pull_request:
+        branches: 
+            - main
+        types: [opened, synchronize, reopened, labeled, unlabeled]
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.run_id }}
```

Do we need the `pull_request` part here?

```suggestion
    group: ${{ github.workflow }}-${{ github.event.number || github.run_id }}
```

Wouldn't work?

Also, can't we just use [GitHub ref](https://docs.github.com/en/actions/using-jobs/using-concurrency#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow)?

```
${{ github.workflow }}-${{ github.ref }}
```

---

In the docs they have mentioned like the `github.ref` will be the `the pull request merge branch` if the workflows are triggered by `pull_request`. So not sure whether the `merge branch` is same for all the commits in a PR. Will test by changing according to the suggestion 👍

---

It worked! Changed according to the suggestion via [c0aff8b](https://github.com/ballerina-platform/module-ballerina-constraint/pull/66/commits/c0aff8b464bd178a7dc492876a42c10618187f18)